src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -765,64 +747,30 @@ public void prestartAllThreads() {
         }
       }
     }
-
-    /**
-     * Getter for the maximum amount of time a low priority task will wait for an available worker.
-     * 
-     * @return currently set max wait for low priority task
-     */
-    public long getMaxWaitForLowPriority() {
-      return maxWaitForLowPriorityInMs;
-    }
-
-    /**
-     * Changes the max wait time for an idle worker for low priority tasks.  Changing this will only 
-     * take effect for future low priority tasks, it will have no impact for the current low priority 
-     * task attempting to get a worker.
-     * 
-     * @param maxWaitForLowPriorityInMs new time to wait for a thread in milliseconds
-     */
-    public void setMaxWaitForLowPriority(long maxWaitForLowPriorityInMs) {
-      ArgumentVerifier.assertNotNegative(maxWaitForLowPriorityInMs, "maxWaitForLowPriorityInMs");
-      
-      this.maxWaitForLowPriorityInMs = maxWaitForLowPriorityInMs;
-    }
-  
+    
     /**
-     * This function REQUIRES that workersLock is synchronized before calling.  It returns an 
-     * available worker if it can get one before the wait time expires.  It will never create 
-     * a new worker.
+     * Call to get (or create, if able to) an idle worker.  If there are no idle workers available 
```

Comment:
Yes, this actually would not create before, but it does not.  That's why it is called "getWorker".  If a worker is not immediately available, and the pool is not maxed out, it will create one.  Because we get the worker before any tasks that means there will be one idle worker thread on creation, before any tasks are submitted to the scheduler (in part because we start the task consumer before any tasks are submitted to the pool as well).

We could delay these thread starts till task execution, but I did not see much reason, this pool is not really designed to be created and tossed away frequently.